fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/BaseDataSourceMetadataSpec.groovy

```diff
@@ -20,82 +20,128 @@ import io.druid.timeline.DataSegment
+ * initiate metadata related resources.
+ * <p>
+ * Take "intervals" as an example
+ * <ul>
+ *     <li>
+ *         you don't need any of the resources(including "intervals") defined in this Spec, you do nothing
+ *     </li>
+ *     <li>
+ *         you need "intervals" in your test, you override childSetupSpec() and call its generation method in your Spec
+ *         as follows
+ *         <pre>
+ *             {@code
+ *             @Override
+ *             def childSetupSpec() {
+ *                 intervals = generateIntervals()
+ *             }
+ *             }
+ *         </pre>
+ *     </li>
+ *     <li>
+ *         you need "intervals" in your test, but you need a different set of intervals:
+ *         <pre>
+ *             {@code
+ *             @Override
+ *             def childSetupSpec() {
+ *                 intervals = generateIntervals()
+ *             }
+ *
+ *             @Override
+ *             Map<String, Interval> generateIntervals() {
+ *                 [
+ *                     "interval1": Interval.parse("2015-01-01T00:00:00.000Z/2015-01-02T00:00:00.000Z"),
+ *                     "interval2": Interval.parse("2015-01-02T00:00:00.000Z/2015-01-03T00:00:00.000Z"),
+ *                     "interval3": Interval.parse("2015-01-03T00:00:00.000Z/2015-01-04T00:00:00.000Z"),
+ *                     "interval123": Interval.parse("2015-01-01T00:00:00.000Z/2015-01-04T00:00:00.000Z")
+ *                 ]
+ *             }
+ *             }
+ *         </pre>
+ *     </li>
+ * </ul>
+ */
+abstract class BaseDataSourceMetadataSpec extends Specification {
     @Shared
-    String tableName = TestDruidTableName.ALL_PETS.asName()
+    DateTimeZone currentTZ
 
     @Shared
-    String intervalString1
-    @Shared
-    String intervalString2
+    String tableName
     @Shared
-    String intervalString12
-
+    Map<String, Interval> intervals
     @Shared
-    Interval interval1
+    RangeSet<DateTime> rangeSet
     @Shared
-    Interval interval2
-    @Shared
-    Interval interval12
-
+    List<TestApiDimensionName> dimensions
     @Shared
-    RangeSet<DateTime> rangeSet12
-
+    List<TestApiMetricName> metrics
     @Shared
-    String version1
+    Integer binversion1
     @Shared
-    String version2
-
+    long size1
     @Shared
-    DataSegment segment1
-
+    long size2
     @Shared
-    DataSegment segment2
-
+    List<String> versions
     @Shared
-    DataSegment segment3
+    List<DataSegment> segments
 
-    @Shared
-    DataSegment segment4
+    def setupSpec() {
+        currentTZ = DateTimeZone.getDefault()
+        DateTimeZone.setDefault(DateTimeZone.UTC)
+        binversion1 = 9
+        size1 = 1024
+        size2 = 512
 
-    @Shared
-    List<DataSegment> segments
+        childSetupSpec()
+    }
 
-    @Shared
-    DateTimeZone currentTZ
+    def childSetupSpec() {}
 
-    @Shared
-    List<String> dimensions123
+    def shutdownSpec() {
```

Comment:
This should be `cleanupSpec`. Also if we're going to have a `childSetupSpec` then we should have a `childCleanupSpec` as well, so that people can cleanup whatever they setup.